### ISSUE TITLE
Short commands often fire unintentionally.

### ISSUE THREAD

**Issue by uchiki** [alterNERDtive/VoiceAttack-profiles#83]
I want you to be able to suppress these commands (eg. cruise, jump) through user-adjustable settings.

In the case of "cruise", "super cruise" and "cruise when ready" remain unchanged and "cruise" is suppressed. With a user setting that is the threshold for "CMDCONFIDENCE".

**Comment by alterNERDtive** [comment]
If they fire unintentionally that usually means your mic and/or speech recognition is not properly set up.

If you want to suppress certain voice triggers you can just create an empty command in your base profile. In your case create a `cruise` command and have it do nothing. If you add an “Other” → “VoiceAttack Action” → “Ignore an Unrecognized Word or Phrase” action it won’t even show up in the log.

**Comment by uchiki** [comment]
I use a voice recognition engine for English for VA,
Since the environmental sound is not only in English, the voice recognition engine is difficult to work well. Especially for commands with short phrases.

And I don't want to make the command completely unrecognizable.

**Comment by alterNERDtive** [comment]
> Since the environmental sound is not only in English, the voice recognition engine is difficult to work well.

Picking up background noise (or having it mess up recognition) would belong into the “mic setup” category. I believe there are some tips on the VA homepage. The general idea is to turn the hardware sensitivity down so you don’t pick up stuff that is far from the mic.

> And I don't want to make the command completely unrecognizable.

If you want to add voice triggers, add a command with those and have it run the command you want. E.g. you can create a command `jump me to super cruise` and have that run the `EliteDangerous.jumpToSupercruise` command.

(Note that with the next major version that will change to `EliteAttack.jumpToSupercruise`).

**Comment by uchiki** [comment]
I think it would be convenient if the user could apply a higher command-specific confidence level, as the user could make some adjustments without rewriting the command.

Anyway, thank you for your polite response.

**Comment by alterNERDtive** [comment]
> I think it would be convenient if the user could apply a higher command-specific confidence level

Yes, it would. The problem is that you lose that on every profile update.

The only thing you can do for that is to set a higher confidence level profile wide in your base profile, that will then also apply to all included profiles. But that would be profile wide, not for a specific command.

**Comment by uchiki** [comment]
> The problem is that you lose that on every profile update.

I don't know how the next major version will change,
The easiest way is to set a variable for the command-specific confidence level in the “startup command”.

**Comment by alterNERDtive** [comment]
> The easiest way is to set a variable for the command-specific confidence level in the “startup command”.

Yeah, but … one for each command? That seems excessive.

**Comment by uchiki** [comment]
I hope only short commands that fire unintentionally are possible.
The command also includes "Dynamic response sections".
Commands that may or may not consist of many syllables are difficult to adjust.

eg.
[jump; engage; get me out; punch it chewie] [and scan;] [when ready;]
[super;] cruise [when ready;]

"cruise" and "jump" are commands with few syllables.

Another solution is to separate commands that have few syllables.

**Comment by alterNERDtive** [comment]
So I’ve slept on this issue and I’ve made the executive decision to not change anything because of it.

If _any_ commands fire unintentionally, that’s a symptom of your mic and/or speech recognition not being set up correctly. And this is _one_ of the symptoms, but you are bound to run into other issues eventually.

So the way to fix this is to fix the underlying problem, not to put band-aids on one of the symptoms. Help for proper setup is available online, e.g. here: https://forum.voiceattack.com/smf/index.php?topic=2667.msg12197#msg12197 Until I set my own stuff up properly and ran some more speech training I had the same issue with the `cruise` command.

Basically it’s

* fix the mic
* train the spech recognition
* adjust command weight and min confidence in VoiceAttack settings

If you can’t fix your setup, I’ve already outlined possible band-aids you can apply yourself. That is the other option beyond fixing the underlying issue. I’m not going to start adding functionality to work around it.

Leaving this open as a reminder to myself to add that voice trigger override stuff to an eventual “how to” wiki entry when I do the documentation overhaul prior to the next major release.

**Comment by alterNERDtive** [comment]
Added to the [troubleshooting section in the docs](https://alternerdtive.github.io/VoiceAttack-profiles/troubleshooting/#voiceattack-does-not-understand-me-mishears-me-fires-random-commands).
